lib/origami-image-set-tools.js

```diff
@@ -51,10 +56,61 @@ module.exports = class OrigamiImageSetTools {
 	 * @returns {Promise} A promise which resolves when the file is written.
 	 */
 	buildImageSetManifestFile() {
+		this.log.info('Building manifest file…');
+		return this.buildImageSetManifest()
+			.then(imageSetManifest => {
+				const filePath = path.join(this.options.baseDirectory, 'imageset.json');
+				const fileContents = JSON.stringify(imageSetManifest, null, '\t');
+				return fs.writeFile(filePath, fileContents);
+			})
+			.then(() => {
+				this.log.info('✔︎ Manifest file saved');
+			})
+			.catch(error => {
+				this.log.error('✘ Manifest file could not be saved');
+				throw error;
+			});
+	}
+
+	/**
+	 * Generate an image set manifest and publish each image to an S3 bucket.
+	 * @param {String} bucket - The name of the S3 bucket.
+	 * @returns {Promise} A promise which resolves when all the images are published.
+	 */
+	publishToS3(bucket) {
+		if (!this.options.awsAccessKey) {
+			return Promise.reject(new Error('No AWS credentials are available'));
+		}
+		const s3 = new AWS.S3({
+			accessKeyId: this.options.awsAccessKey,
+			secretAccessKey: this.options.awsSecretKey,
+			params: {
+				Bucket: bucket
+			}
+		});
+		const s3BasePath = `${this.scheme}/v${semver.major(this.options.version)}`;
 		return this.buildImageSetManifest().then(imageSetManifest => {
-			const filePath = path.join(this.options.baseDirectory, 'imageset.json');
-			const fileContents = JSON.stringify(imageSetManifest, null, '\t');
-			return fs.writeFile(filePath, fileContents);
+			return Promise.all(imageSetManifest.images.map(image => {
+				const fullPath = path.resolve(this.options.baseDirectory, image.path);
+				const s3Path = `${s3BasePath}/${image.name}.${image.extension}`;
+				const s3Config = {
+					ACL: 'public-read',
+					Body: fs.createReadStream(fullPath),
+					ContentType: mime.lookup(fullPath),
```

Comment:
yep there's already a whitelist – the function that builds the manifest uses [this array](https://github.com/Financial-Times/origami-image-set-tools/blob/1388562ded7b78932eb966a538e312e7615b4c92/lib/origami-image-set-tools.js#L137) to filter files.